packages/manager/apps/container/src/hooks/agreements/useAgreementsUpdate.tsx

```diff
@@ -0,0 +1,12 @@
+import { DefinedInitialDataOptions, useQuery } from '@tanstack/react-query';
+import fetchAgreementsUpdates from '@/api/agreements';
+import { Agreements } from '@/types/agreements';
+
+const useAgreementsUpdate = (options?: Partial<DefinedInitialDataOptions<Agreements[]>>) =>
+  useQuery({
+    ...options,
+    queryKey: ['agreements'],
+    queryFn: fetchAgreementsUpdates,
+  });
```

Comment:
Same as https://github.com/ovh/manager/pull/14041/files#r1901556827 + Query key is too subtle with no info regarding the applied filters. It can easily induce errors if same key is used with different filters.
```suggestion
const usePendingAgreements = (options?: Partial<DefinedInitialDataOptions<Agreements[]>>) =>
  useQuery({
    ...options,
    queryKey: ['agreements'],
    queryFn: fetchAgreementsUpdates,
  });
```